src/Microsoft.AspNetCore.Blazor.Server/Circuits/RemoteRenderer.cs

```diff
@@ -87,7 +89,15 @@ public void Dispose()
         /// <inheritdoc />
         protected override void UpdateDisplay(in RenderBatch batch)
         {
-            var task = _client.SendAsync("JS.RenderBatch", _id, batch);
+            // Send the render batch to the client
+            // Note that we have to capture the data as a byte[] synchronously here, because
+            // SignalR's SendAsync can wait an arbitrary duration before serializing the params.
+            // The RenderBatch buffer will get reused by subsequent renders, so we need to
+            // snapshot its contents now.
+            // TODO: Consider using some kind of array pool instead of allocating a new
```

Comment:
I think we discussed this before. It would actually be very easy if not for the fact that the message pack library is the one creating the arrays. I think we need to be really concerned about LOH allocations but not in 0.7

Reply:
Reopened #1132